ouroboros-consensus/docs/diffusion-pipelining-design/block-diffusion-pipelining.tex

```diff
@@ -0,0 +1,554 @@
+
+Fortunately this problem does not appear insurmountable. We can still pin the
+blame for a bad block on the issuer of the block. The block issuer is typically
+not a direct peer therefore we cannot use disconnection as a solution. If however, we simply
+employ some state to temporarily store the block issuer details, then we can use that reference
+to not consider any more blocks signed by this issuer in this slot. They
+have demonstrated themselves to be adversarial by signing a bad block, and we
+shouldn't spend any further resources validating blocks from them in the short term. 
+Note that for this purpose we must identify the block issuer by the hot key certificate and 
+not the `cold' permanent identity. This is due to the fact that a compromised hot key could 
+be used to sign bad blocks, and that ought to still be a situation from which a block producer can 
+recover from by signing a new certificate with a higher issue number.
+
+It appears that it would be sufficient to hold a state of bad block issuers
+(identified by certificate) \emph{only until} the current chain is successfully
+extended. This approach would avoid considering more than a single bad block from any
+recent issuer. The number of recent adversarial issuers (and thus the size of
+the tracking set) is bounded by the leadership schedule and the honest stake
+assumption. The certificate counters are bounded (or can be bounded) as
+described in \cref{many-valid-block-headers}.
+
+\pagebreak
+
+\section{Implementation changes}
+The implementation changes required to deliver on the approach discussed in this document
+are all to be made within the Consensus layer. The discussion in this section pertains specifically 
+to the "non-aggressive" diffusion pipelining variant. The specific novel behaviour is that nodes now
+forward \emph{one unvalidated block}, so that multiple nodes can download and validate the latest 
+block \emph{concurrently}. To begin, let us remind ourselves how Cardano currently diffuses data:
+
+\begin{itemize}
+  \item With Ouroboros Praos, we want the block minted in a given slot to diffuse to the leader of the next slot early enough that they in-turn can extend the chain on-top of it.
+  \item With Praos, a node will adopt the chain which is most dense and whose intersection with the previous "best" known chain is less than $k$ blocks, where $k$ is currently set to 2160.
+  \item Cognisant of the above, the process continues as follows:
+  \begin{enumerate}
+    \item A node will notify interested peers that a new block is available by posting the new header to \lstinline{ChainSync} mini-protocol, after block validation. 
+    \item After validation of the header locally, the downstream peer will request/pull the corresponding block body.
+    \item The block is validated, and if it satisfies checks resulting in a new longer/better chain, it will be selected as the head of our "new" chain.
```

Comment:
The blocks might arrive out of order, in which case a successful block validation might change the current chain, or extend it, but it does not necessarily mean that the newly validated block is the "head" of our new chain. 

Also about the notation, I think we might prefer "tip" as opposed to "head" to refer to the newest element in a chain.